Add Time(), Before() and After() methods to Date type.

pnawalramka opened the pull request:
Adds helpful utility methods `Time()`, `Before()` and `After()` to `Date` type. The latter two are meant to provide similar functionality as standard library's [time.Before](https://pkg.go.dev/time#Time.Before) and [time.After](https://pkg.go.dev/time#Time.After) methods.

Added tests!

mpoindexter commented:
Do Before and After really need to operate on `time.Time`s? I feel like you could do a lexical comparison given the date format and this would mean that you could compare without returning an `error` object because you can skip the time parse step.

I also don't love the addition of the `Time` method - logically speaking a Date encodes a calendar date, not a specific moment in time. For example, once you convert it to a `Time`, it's now a fixed instant, that may actually be different dates in different timezones. This is not _wrong_, but I've found that converting between instants and calendar dates to be a good source of errors, so if this is not needed except for the `Before` and `After` methods I would leave it out given the above comment.

pnawalramka commented:
Good point regarding dependency on `time.Time`. I converted the methods to perform lexical comparison. The error condition still holds though, because there is no format validation on instantiation.

Regarding the `Time()` method, I found that most users of the library end up converting the `Date` instance to `time.Time` anyway. That said, I guess the motivation is to use functionalities similar to those being added in this PR. I think it's reasonable to not make assumptions around timezone and location and leave it up to the caller to make necessary conversions. Removed the method from this PR.
